static/app/components/events/interfaces/crashContent/stackTrace/content.tsx

```diff
@@ -49,11 +50,93 @@ class Content extends Component<Props, State> {
     expandFirstFrame: true,
   };
 
+  constructor(props) {
+    super(props);
+    this.state.toggleFrameMap = this.setInitialFrameMap();
+  }
+
   state: State = {
     showingAbsoluteAddresses: false,
     showCompleteFunctionName: false,
+    toggleFrameMap: {},
+  };
+
+  setInitialFrameMap(): {[frameIndex: number]: boolean} {
+    const {data} = this.props;
+    const indexMap = {};
+    (data.frames ?? []).forEach((frame, frameIdx) => {
+      const nextFrame = (data.frames ?? [])[frameIdx + 1];
+      const repeatedFrame =
```

Comment:
It looks like this logic is repeated a few times in the file. Can you extract this out to a function like `isRepeatedFrame(frame: Frame, nextFrame: Frame)`?